src/platform/Ameba/AmebaConfig.cpp

```diff
@@ -90,6 +90,8 @@ CHIP_ERROR AmebaConfig::ReadConfigValue(Key key, bool & val)
 
     val = (intVal != 0);
 
+    free(_namespace);
```

Comment:
namespace is still leaked in the early return of `_name == NULL` check.

Use a scoped/unique_ptr instead to ensure this does not happen and is provable correct. RAII is prefferable in C++ code.

Reply:
Are we copying these strings for some const correctness thing?

If it's truly necessary can we at least factor out the malloc/strcpy into a helper function...?